liborbum/src/Controller/Ee/Vpu/Vif/CVif.cpp

```diff
@@ -95,48 +95,101 @@ void CVif::INSTRUCTION_UNSUPPORTED(VifUnit_Base* unit, const VifcodeInstruction
     throw std::runtime_error("VIFcode CMD field was invalid! Please fix.");
 }
 
+// Refer to EE Users Manual pg 103.
 void CVif::NOP(VifUnit_Base* unit, const VifcodeInstruction inst)
 {
+    // nothing to do
 }
 
+// Refer to EE Users Manual pg 104.
 void CVif::STCYCL(VifUnit_Base* unit, const VifcodeInstruction inst)
 {
+    // Writes CODE.IMMEDIATE to CYCLE
+    uword immediate = unit->code.IMMEDIATE.extract_from(unit->code.read_uword());
+    unit->cycle.write_uword(immediate);
 }
 
+// Refer to EE Users Manual pg 105.
 void CVif::OFFSET(VifUnit_Base* unit, const VifcodeInstruction inst)
 {
+    // VIF1 only
+    if (unit->core_id != 1)
+        return;
```

Comment:
If you could also log a message when the core_id is not 1 (for all VIF1 only instructions), that would be great. I have a feeling there will be bugs with this, the docs are not always 100% correct.

Reply:
To log a message you can type :
```
#include <boost/format.hpp>
#include "Core.hpp"
BOOST_LOG(Core::get_logger()) << str(boost::format("Warning: VIF1 only instruction, called by VIF%d") % unit->core_id);
```

Look at the other code for more examples if you need to.